### ISSUE TITLE
Exception when Change TTL 

### ISSUE THREAD

**Issue by boliza** [thinkaurelius/titan#980]
My Schema:

``` java
TitanSchemaType bloggerVertexLabel = mgmt.makeVertexLabel("user").make();
PropertyKey userId = mgmt.makePropertyKey("userId").dataType(String.class).make();
TitanGraphIndex bloggerIdIndex = mgmt.buildIndex("userIdIndex", Vertex.class).indexOnly(bloggerVertexLabel).addKey(userId).unique().buildCompositeIndex();
mgmt.setConsistency(bloggerIdIndex, ConsistencyModifier.LOCK);
mgmt.addIndexKey(getExternalIndex(Vertex.class), userId, Mapping.STRING.getParameter());
PropertyKey key = mgmt.makePropertyKey("userName").dataType(String.class).make();
mgmt.setTTL(key, 20, TimeUnit.SECONDS);
```

After my schema is created,When change the TTL I see an exception becasue of this code in AbstractVertex line 90.

**Comment by boliza** [comment]
Seems to change the assert to if condition could avoid this

**Comment by dalaro** [comment]
I adapted the last two lines from your example.  I ran that on titan09 in gremlin.sh (at d704116e9577867e9cff5d962bafd02e43f405c2, using Cassandra + ES).  It seemed OK:

``` groovy
gremlin> g = TitanFactory.open('conf/titan-cassandra.properties')
==>standardtitangraph[cassandrathrift:[127.0.0.1]]
gremlin> mgmt = g.openManagement()
==>com.thinkaurelius.titan.graphdb.database.management.ManagementSystem@6631cb64
gremlin> userName = mgmt.makePropertyKey("userName").dataType(String.class).make()
==>userName
gremlin> mgmt.setTTL(userName, 20, java.util.concurrent.TimeUnit.SECONDS)
==>null
gremlin> mgmt.commit()
==>null
```

Then adapted your whole example in gremlin.sh, wondering if there was some obscure interaction between the last two lines and the foregoing ones that you included.  I still don't see anything broken:

``` groovy
gremlin> g = TitanFactory.open('conf/titan-cassandra-es.properties')
==>standardtitangraph[cassandrathrift:[127.0.0.1]]
gremlin> mgmt = g.openManagement()
==>com.thinkaurelius.titan.graphdb.database.management.ManagementSystem@da4cf09
gremlin> bloggerVertexLabel = mgmt.makeVertexLabel("user").make()
==>user
gremlin> userId = mgmt.makePropertyKey("userId").dataType(String.class).make()
==>userId
gremlin> bloggerIdIndex = mgmt.buildIndex("userIdIndex", Vertex.class).indexOnly(bloggerVertexLabel).addKey(userId).unique().buildCompositeIndex()
==>userIdIndex
gremlin> mgmt.setConsistency(bloggerIdIndex, ConsistencyModifier.LOCK)
==>null
gremlin> mgmt.buildIndex("vsearch", Vertex.class).buildMixedIndex("search")
==>vsearch
gremlin> userName = mgmt.makePropertyKey("userName").dataType(String.class).make()
==>userName
gremlin> mgmt.setTTL(userName, 20, java.util.concurrent.TimeUnit.SECONDS)
==>null
gremlin> mgmt.commit()
==>null
```

Can you please be more specific?

**Comment by boliza** [comment]
I using titan0.5.4,have not test in titan09.I will try again

**Comment by dalaro** [comment]
I'll try that.

**Comment by dalaro** [comment]
Seems OK on the 0.5.4 tag:

``` java
gremlin> g = TitanFactory.open('conf/titan-cassandra-es.properties')
1922 [main] INFO  com.thinkaurelius.titan.core.util.ReflectiveConfigOptionLoader  - Loaded and initialized config classes: 12 OK out of 12 attempts in Duration[42 ms]
2037 [main] INFO  com.thinkaurelius.titan.graphdb.configuration.GraphDatabaseConfiguration  - Set cluster.partition=false from store features
2076 [main] INFO  com.thinkaurelius.titan.graphdb.configuration.GraphDatabaseConfiguration  - Set default timestamp provider MICRO
2082 [main] INFO  com.thinkaurelius.titan.diskstorage.cassandra.thrift.CassandraThriftStoreManager  - Closed Thrift connection pooler.
2087 [main] INFO  com.thinkaurelius.titan.graphdb.configuration.GraphDatabaseConfiguration  - Generated unique-instance-id=7f0001012002-bluebox1
2097 [main] INFO  com.thinkaurelius.titan.diskstorage.Backend  - Configuring index [search]
2278 [main] INFO  org.elasticsearch.plugins  - [Jimmy Woo] loaded [], sites []
2755 [main] INFO  com.thinkaurelius.titan.diskstorage.es.ElasticSearchIndex  - Configured remote host: 127.0.0.1 : 9300
3574 [main] INFO  com.thinkaurelius.titan.diskstorage.Backend  - Initiated backend operations thread pool of size 8
4055 [main] INFO  com.thinkaurelius.titan.diskstorage.Backend  - Configuring total store cache size: 869461720
4384 [main] INFO  com.thinkaurelius.titan.diskstorage.log.kcvs.KCVSLog  - Loaded unidentified ReadMarker start time Timepoint[1429844548865000 μs] into com.thinkaurelius.titan.diskstorage.log.kcvs.KCVSLog$MessagePuller@2f39fb7f
==>titangraph[cassandrathrift:[127.0.0.1]]
gremlin> mgmt = g.getManagementSystem()
==>com.thinkaurelius.titan.graphdb.database.management.ManagementSystem@747d71e2
gremlin> bloggerVertexLabel = mgmt.makeVertexLabel("user").make()
==>user
gremlin> userId = mgmt.makePropertyKey("userId").dataType(String.class).make()
==>userId
gremlin> bloggerIdIndex = mgmt.buildIndex("userIdIndex", Vertex.class).indexOnly(bloggerVertexLabel).addKey(userId).unique().buildCompositeIndex()
==>com.thinkaurelius.titan.graphdb.database.management.TitanGraphIndexWrapper@71cce1e
gremlin> mgmt.setConsistency(bloggerIdIndex, ConsistencyModifier.LOCK)
==>null
gremlin> mgmt.buildIndex("vsearch", Vertex.class).buildMixedIndex("search")
==>com.thinkaurelius.titan.graphdb.database.management.TitanGraphIndexWrapper@78e07d6c
gremlin> userName = mgmt.makePropertyKey("userName").dataType(String.class).make()
==>userName
gremlin> mgmt.setTTL(userName, 20, java.util.concurrent.TimeUnit.SECONDS)
==>null
gremlin> mgmt.commit()
==>null
gremlin>
```

**Comment by boliza** [comment]
seems fixed in commit df42713
